catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/services/ingestionpipelines/IngestionPipelineResource.java

```diff
@@ -520,6 +520,24 @@ public Response getRESTStatus(@Context UriInfo uriInfo, @Context SecurityContext
     return pipelineServiceClient.getServiceStatus();
   }
 
+  @GET
+  @Path("/ip")
+  @Operation(
+      operationId = "checkAirflowHostIp",
+      summary = "Check the Airflow REST host IP",
+      tags = "IngestionPipelines",
+      description = "Check the Airflow REST host IP",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "Pipeline Service host IP",
+            content = @Content(mediaType = "application/json"))
+      })
+  public Response getHostIp(@Context UriInfo uriInfo, @Context SecurityContext securityContext) {
+    Map<String, String> hostIp = pipelineServiceClient.getHostIp();
+    return Response.ok(hostIp, MediaType.APPLICATION_JSON_TYPE).build();
+  }
```

Comment:
The `IngestionPipelineResource` class looks like a proxy to communicate with a `PipelineServiceClient`. Do you think it is a good idea to have a task to start thinking about isolating it in a different `Resource` class, e.g., `PipelineServiceClientResource` or something similar?

Reply:
IMO once we start the exercise about integrating Argo, it will be a good opportunity to see which parts can be bundled/unbundled, but yes agree that we are currently repeating some stuff